cmd/diff/cmddiff.go

```diff
@@ -80,17 +80,12 @@ func Initialize(o *diff.DiffOptions, f util.Factory, args []string) (func(), err
 		return cleanupFunc, err
 	}
 
-	o.DiscoveryClient, err = f.ToDiscoveryClient()
-	if err != nil {
-		return cleanupFunc, err
-	}
-
 	o.DynamicClient, err = f.DynamicClient()
 	if err != nil {
 		return cleanupFunc, err
 	}
 
-	o.DryRunVerifier = resource.NewDryRunVerifier(o.DynamicClient, o.DiscoveryClient)
+	o.DryRunVerifier = resource.NewQueryParamVerifier(o.DynamicClient, f.OpenAPIGetter(), resource.QueryParamDryRun)
```

Comment:
This change is similar to https://github.com/kubernetes/kubectl/commit/7e681490ec30a64626b359b7346d891240155794